### DISCUSSION TITLE
Is ASProgrammer compatible with Arduino Uno?

### DISCUSSION THREAD

**Discussion by mitko-kamburow** [nofeletru/UsbAsp-flash#64] [Q&A] [answered]
The title says it. Also is there a diagram?

**Comment by nofeletru** [comment]
firmware/asprog/defines.h contain pin defination.

Look at asprog.ino #define UART_SPEED 1000000 
settings.xml also have <options arduino_baudrate="1000000"/>
There is no baudrate option in software IU.

**Reply by mitko-kamburow** [reply]
What if I'm using a SPI chip?

**Comment by nofeletru** [comment, marked as answer]
![arduinoSPI25](https://user-images.githubusercontent.com/12898237/161415927-892058b7-ff9e-4cee-a81c-67443ddc03af.gif)

Arduino pin numbers in parentheses.
Baudrate need to be same in program and scetch.

**Reply by mitko-kamburow** [reply]
Thanks!

**Reply by Tomblarom** [reply]
Thanks @nofeletru. This made it very obvious to me. :)

**Comment by Gouster4** [comment]
I had an CH341a, and it stopped working in middle of programming flash on my NVR. Found one Arduino UNO laying around, so i soldered clips on it, and continued modifying telnet passwords on all telnet enabled devices in my household.

But i ordered 2nd CH341a. Its cool that Arduino can flash SPI, it was an life saver. But it too slow compared to CH341a.
